sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -1640,6 +1904,34 @@ class StreamingOuterJoinSuite extends StreamingJoinSuite {
       }
     }
   }
+
+  test("SPARK-51779 - " +
```

Comment:
Shall we modify the existing test `left semi early state exclusion on left` to perform restart instead of adding a new test?